dashboard/test/ui/features/step_definitions/steps.rb

```diff
@@ -771,6 +774,15 @@ def disabled?(selector)
   expect(disabled?(selector)).to eq(false)
 end
 
+Then /^I wait (?:up to (\d+) second[s]? )?until "([^"]*)" is (not )?disabled$/ do |wait, selector, negation|
```

Comment:
This one could use a brief top-level comment:
* if `up to N seconds` is specified, succeeds when the element becomes enabled/disabled or N seconds have passed
* otherwise, succeeds when the element becomes enabled/disabled, or FAILS if `DEFAULT_WAIT_TIMEOUT` seconds pass first

I know you mention this inline below, but reading that comment makes me question the rest of the contract so it would be nice to have it all written in one place.